mlflow/java/client/src/main/java/org/mlflow/tracking/MlflowClient.java

```diff
@@ -119,42 +119,104 @@ public RunInfo createRun(CreateRun request) {
+   *                     If null, the result will be equivalent to having an empty search filter.
    *
-   * @return A list of all Runs that satisfy search filter.
+   * @return A list of all RunInfos that satisfy search filter.
    */
-  public List<Run> searchRuns(List<String> experimentIds, String searchFilter) {
+  public List<RunInfo> searchRuns(List<String> experimentIds, String searchFilter) {
     return searchRuns(experimentIds, searchFilter, ViewType.ACTIVE_ONLY);
   }
 
+  /**
+   * Return RunInfos from provided list of experiments that satisfy the search query.
+   * @deprecated As of 1.1.0 - please use {@link #searchRuns(List, String, ViewType, int)} or 
+   *                    similar that returns a page of Run results.
+   *
+   * @param experimentIds List of experiment IDs.
+   * @param searchFilter SQL compatible search query string. Format of this query string is
+   *                     similar to that specified on MLflow UI.
+   *                     Example : "params.model = 'LogisticRegression' and metrics.acc != 0.9"
+   *                     If null, the result will be equivalent to having an empty search filter.
+   * @param runViewType ViewType for expected runs. One of (ACTIVE_ONLY, DELETED_ONLY, ALL)
+   *                    If null, only runs with viewtype ACTIVE_ONLY will be searched.
+   *
+   * @return A list of all RunInfos that satisfy search filter.
+   */
+  public List<RunInfo> searchRuns(List<String> experimentIds,
+                              String searchFilter,
+                              ViewType runViewType) {
+    SearchRuns.Builder builder = SearchRuns.newBuilder()
+            .addAllExperimentIds(experimentIds);
+
+    if (searchFilter != null) {
+      builder.setFilter(searchFilter);
+    }
+    if (runViewType != null) {
+      builder.setRunViewType(runViewType);
+    }
+    SearchRuns request = builder.build();
+    String ijson = mapper.toJson(request);
+    String ojson = sendPost("runs/search", ijson);
+    return mapper.toSearchRunsResponse(ojson).getRunsList().stream().map(Run::getInfo)
+            .collect(Collectors.toList());
+  }
+
   /**
    * Return runs from provided list of experiments that satisfy the search query.
    *
    * @param experimentIds List of experiment IDs.
    * @param searchFilter SQL compatible search query string. Format of this query string is
    *                     similar to that specified on MLflow UI.
    *                     Example : "params.model = 'LogisticRegression' and metrics.acc != 0.9"
+   *                     If null, the result will be equivalent to having an empty search filter.
    * @param runViewType ViewType for expected runs. One of (ACTIVE_ONLY, DELETED_ONLY, ALL)
-   *                    Defaults to ACTIVE_ONLY.
+   *                    If null, only runs with viewtype ACTIVE_ONLY will be searched.
+   * @param maxResults Maximum number of runs desired in one page. Defaults to 1000 runs per page.
```

Comment:
What does it mean here for the default to 1000 runs per page?

Reply:
Ah sorry. Meant to get rid of that. It should always be defined